.github/workflows/paralleltest-docker.yml

```diff
@@ -0,0 +1,113 @@
+name: Run AQAvit Paralleltest docker
+
+on:
+  workflow_dispatch: # Allows the job to be manually triggered
+    inputs:
+      customizedSdkUrl:
+         description: SDK URL
+         required: true
+      testImage:
+        description: Native Libs URL
+        required: true
+      aqatest-repo:
+        description: adoptium/aqa-tests repository
+        required: true
+        type: string
+        default: 'adoptium/aqa-tests:v0.9.6-release'
+      containerRegistry:
+        description: Container registry
+        required: true
+
+jobs:
+  setup-parallel-docker:
+    runs-on: ubuntu-latest
+    container:
+      image: ${{ inputs.containerRegistry }}
+      env:
+        USE_TESTENV_PROPERTIES: true
+      options: --cpus 2
+      credentials:
+        username: ${{ secrets.ACR_TOKEN_NAME }}
+        password: ${{ secrets.AZURE_ACR_TOKEN_PASS1 }}
+    strategy:
+      fail-fast: true
+      matrix:
+        suite: [functional, openjdk, system, perf]
+    steps:
+
+      - name: Setup environment
+        working-directory: ${{ github.workspace }}
+        shell: bash
+        run: |
+          curl -s -o ${{ inputs.system }}-binaries.tar.gz "${{ secrets.BASE_CONTAINER_URL }}${{ inputs.customizedSdkUrl }}?${{ secrets.SAS_TOKEN }}"
+          mkdir tmp && tar -xzf ${{ inputs.system }}-binaries.tar.gz -C ./tmp
+          jdk_directory=$(ls ./tmp)
+          echo "TEST_JDK_HOME=${{ github.workspace }}/tmp/$jdk_directory" >> $GITHUB_ENV
+
+      - uses: adoptium/run-aqa@v2.0.1
+        with:
+          jdksource: 'customized'
+          aqa-testsRepo: ${{ inputs.aqatest-repo }}
+          build_list: ${{ matrix.suite }}
+          target: ${{ matrix.suite }} # parallellist to generate
+          run_parallel: true
+          num_machines: 4
+
+      - name: Archive parallelList - ${{ matrix.suite }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.suite }}-parallelList.mk
+          path: ${{ github.workspace }}/aqa-tests/TKG/parallelList.mk
+
+  run-aqa-parallel-docker:
+    runs-on: ubuntu-latest
+    needs: setup-parallel-docker
+    container:
+      image: ${{ secrets.CONTAINER_REGISTRY }}

```

Comment:
Just curious, why is "containerRegistry" both an input and a secret?

Reply:
Ah, secret needs to be removed. YAML should be fine.